### DISCUSSION TITLE
Officially recommended etcd python packages in 2024?

### DISCUSSION THREAD

**Discussion by Machoney** [etcd-io/etcd#18211] [Q&A] [answered]
As a follow up of #13285, I have looked at recommended python packages and looks like all of them are not currently under active maintenance.

As of the day of June 21, 2024:

[kragniz/python-etcd3](https://github.com/kragniz/python-etcd3) - Client for v3
Last commit: Aug 17 2020, Last Release: Mar 31 2020

[jplana/python-etcd](https://github.com/jplana/python-etcd) - Supports v2
Last commit: Oct 31 2023, Last Release Mar 3,** 2017**

[russellhaering/txetcd](https://github.com/russellhaering/txetcd) - a Twisted Python library
Last Commit: Oct 28 2013, Last Release: Oct 1, 2013

[cholcombe973/autodock](https://github.com/cholcombe973/autodock) - A docker deployment automation tool
Last Commit: Mar 26 2014

[lisael/aioetcd](https://github.com/lisael/aioetcd) - (Python 3.4+) Asyncio coroutines client (Supports v2)
Last Commit: Dec 18, 2014

[txaio-etcd](https://github.com/crossbario/txaio-etcd) - Asynchronous etcd v3-only client library for Twisted (today) and asyncio (future)
Last Commit: Feb 22, 2019, Last Release: Oct 11, 2018

[dims/etcd3-gateway](https://github.com/dims/etcd3-gateway) - etcd v3 API library using the HTTP grpc gateway
Last Commit: Aug 13, 2020, Last Release: Jun 30, 2020

[aioetcd3](https://github.com/gaopeiliang/aioetcd3) - (Python 3.6+) etcd v3 API for asyncio - Last Commit: Sep 30, 2020
[Revolution1/etcd3-py](https://github.com/Revolution1/etcd3-py) - (python2.7 and python3.5+) Python client for etcd v3, using gRPC-JSON-Gateway - Last Commit: May 10, 2022, Last Release: May 10, 2019

Packages that are not maintained anymore, should not be officially supported as a preferred packages.

**Comment by jmhbnz** [comment, marked as answer]
Hi @Machoney - Thanks for raising this discussion. From a project perspective there are no officially supported python clients. 

The officially maintained clients are:
- https://pkg.go.dev/go.etcd.io/etcd/client/v3
- https://github.com/etcd-io/jetcd

The projects listed under https://etcd.io/docs/v3.5/integrations/#python are for community reference and knowledge and the following disclaimer from the top of the page applies:

> Note that third-party libraries and tools (not hosted on https://github.com/etcd-io) mentioned below are not tested or maintained by the etcd team. Before using them, users are recommended to read and investigate them.

**Comment by atline** [comment]
Does anyone ever used https://opendev.org/openstack/etcd3gw (PS: I don't know how to communicate with these developers...)? It looks this is one which actively maintained, but I'm not sure it's an alternative of existed python library. Anyone could give some inspire if we can use this?
